File: ports/java/src/gui/layout/BcsLayout.java
package bcs.gui.layout;

import bcs.gui.widgets.BcsWidget;
import java.util.ArrayList;
import java.util.List;

// BcsLayout governs spatial organization of UI components
public class BcsLayout {
    protected final List<BcsWidget> children;
    public int margin;
    public int spacing;

    public BcsLayout() {
        this.children = new ArrayList<>();
        this.margin = 0;
        this.spacing = 5;
    }

    public void addWidget(BcsWidget w) {
        this.children.add(w);
    }

    public void removeWidget(BcsWidget w) {
        this.children.remove(w);
    }
}

File: ports/csharp/src/gui/layout/BcsLayout.cs
using System.Collections.Generic;
using BCS.Gui.Widgets;

namespace BCS.Gui.Layout {
    // BcsLayout governs spatial organization of UI components
    public class BcsLayout {
        protected List<BcsWidget> _children;
        public int Margin { get; set; }
        public int Spacing { get; set; }

        public BcsLayout() {
            _children = new List<BcsWidget>();
            Margin = 0;
            Spacing = 5;
        }

        public void AddWidget(BcsWidget w) {
            _children.Add(w);
        }

        public void RemoveWidget(BcsWidget w) {
            _children.Remove(w);
        }
    }
}
